alivee/general_utils/requests/google/place_search/utils.py

```diff
@@ -0,0 +1,287 @@
+		'''
+		self.location = location
+		try:
+			self.client = googlemaps.Client(key)
+		except:
+			raise InvalidRequestAPIKeyException(key=key)
+
+	def _getPlaceFilterKwargs(self, place_type=None, radius=None, keyword=None, name=None, rank_by=None, 
+		page_token=None):
+		'''
+		Private helper method to pre-check and create the filter kwargs in place search query, follow format of
+		https://github.com/googlemaps/google-maps-services-python/blob/master/googlemaps/places.py#L199
+
+		return: A map includes all params in place search query.
+
+		NOTE: Pre-check is needed to avoid obviously invalid API call, thus saving query count.
+		'''
+		filterKwargs = dict(location=self.location)
+
+		if place_type:
+			# Check then include `place type` info.
+			if not place_type in PLACE_SEARCH_REQUEST_PLACE_TYPES:
+				raise InvalidRequestPlaceTypeException(placeType=place_type)
+			filterKwargs['type'] = place_type
+
+		if radius:
+			# Check then include `radius` info.
+			if radius > MAX_PLACE_SERACH_RADIUS_IN_METERS:
+				raise ExceedMaximumRequestRadiusException(radius=radius, maxRadius=MAX_PLACE_SERACH_RADIUS_IN_METERS)
+			filterKwargs['radius'] = radius
+
+		if keyword:
+			# Check then include `keyword` info.
+			filterKwargs['keyword'] = keyword
+
+		if name:
+			# Check then include `name` info.
+			filterKwargs['name'] = name
+
+		if rank_by:
+			# Check then include `rank_by` info.
+			if rank_by == PlaceSearchRankByMethod.DISTANCE:
+
+				if radius:
+					raise ConflictRequestRadiusRankByException()
+
+				if (not place_type) and (not keyword) and (not name):
+					raise IncompleteRequestRankByRelatedParamsException()
+
+			filterKwargs['rank_by'] = rank_by
+
+		if page_token:
+			# Check then include `page_token` info.
+			filterKwargs['page_token'] = page_token
+
+		return filterKwargs
+
+	def _checkReponseStatus(self, response):
+		'''
+		Private helper method to check if the status of google place search response is valid.
+
+		param response: Response of google place search query, notice `googlemaps` package already parses the 
+		JSON format response to `dict - list` format for us.
+		return: `True` means the status in response is valid, otherwise an InvalidResponseStatusException 
+		will be raised.
+		'''
+		response = response or {}
+		status = response.get('status')
+
+		if status not in [PlaceSearchResponseStatus.OK.value, PlaceSearchResponseStatus.ZERO_RESULTS.value]:
+			raise InvalidResponseStatusException(status=status)
+
+		return True
+
+	def _parseNearbySearchResponse(self, response):
+		'''
+		Private helper method to parse response of nearby search.
+
+		param response: Response of google place nearby search query.
+		return: A list of all parsed results for place search query.
+
+		NOTE: Retry for next page fetch query happens here while retry for base place search query 
+		should be handled in caller's method.
+		'''
+		allParsedResults = []
+
+		try:
+			self._checkReponseStatus(response=response)
+		except:
```

Comment:
Oh yeah, previously wanted to raise a diff type exception then changed my mind, let me update thanks.